syntax = "proto3";

package contentprovider;

service SuzukiKasamiService {
  rpc RequestToken(Request) returns (Ack) {}
  rpc ReceiveToken(Token) returns (Ack) {}
}

message Request {
  int32 node_id = 1;
  int32 request_number = 2;
}

message Token {
  int32 holder_id = 1;
  repeated int32 RN = 2;
}

message Ack {}